### PR TITLE
build: adjust how find order is prioritized

### PR DESCRIPTION
**Pull request by dundargoc**
Ensure bundled libraries and include directories are always searched
first before any others. This will provide a more consistent experience
as the search order of the builtin find_ functions can vary depending on
system. This should make the build process faster when building with
bundled deps as we limit the search to only the .deps directory.
Separating the search between .deps and everything makes debugging
find_-related problems simpler if you need to check how dependencies are
found.

For libraries, we divide the search process into the following order:
1. Only search in .deps directory and only search for static libraries.
2. Only search in .deps directory and search for all libraries.
3. Search everywhere and search for all libraries.

Make an exception for FindLibintl.cmake as changing the search order
seems to break some tests on macos.

### CODE REVIEW

File: cmake/FindLibintl.cmake [hunk 1]
```diff
@@ -18,9 +18,8 @@ find_path(LIBINTL_INCLUDE_DIR
     PATH_SUFFIXES gettext
 )
 
-find_library(LIBINTL_LIBRARY
-    NAMES intl libintl
-)
+find_library(LIBINTL_LIBRARY NAMES libintl${CMAKE_STATIC_LIBRARY_SUFFIX})
```
Comment (dundargoc) [inline cmake/FindLibintl.cmake:21]: Just write `libintl.a`, this won't work on windows anyways due to the `lib` prefix.

Comment (dundargoc) [reply cmake/FindLibintl.cmake:21]: TODO: ayo hold up. Where does the static libintl even come from? I see we don't even build it on non-windows.

**Comment by dundargoc** [comment]
TODO: maybe just skip the static/dynamic entirely and focus on limiting search paths to our deps

**Comment by dundargoc** [comment]
I will accept bikeshedding on the function names lol. Couldn't come up with anything better.

File: cmake/Find.cmake [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Functions to aid the built-in find_ functions
+
+# Same as find_path, but always search in .deps directory first and then everything else.
+function(find_path2)
```
Comment (justinmk) [inline cmake/Find.cmake:4]: This is a good name imo. The "2" is a strong hint that it's similar to the standard find_path. Using a different name  would reduce intuition while gaining nothing.

**Review by justinmk — APPROVED** [review]
I'm surprised that the search path order isn't controllable in some other way. But I'm sure you already consider that.

**Comment by dundargoc** [comment]
I'm pretty much 100% sure this is the only way to ensure static libs are prioritized over shared libs. As for prioritizing .deps over system library I'm not sure. I couldn't find anything that seemed to solve this well. If we come up with something more graceful later we should use that.

**Comment by justinmk** [comment]
Yeah, this is isolated and easy to revert if we find a better way later.

**Pull request merged**